src/paho/mqtt/client.py

```diff
@@ -19,6 +19,7 @@
 
 import base64
```

Comment:
It could be a good idea to consider `from __future__ import annotations`, so you wouldn't need the stringly-typed future references (e.g. `Callable["Client"]` could be `Callable[Client]`).

That will also allow you to use modern type annotation syntax even if the target Python version might not necessarily support it (e.g. `Foo | None` instead of `Optional[Foo]`), as type checkers do understand the syntax. Ruff has a rule or two to do that auto-conversion.

Reply:
I didn't known this one. It's really cool.